packages/itwin/grouping-mapping-widget/src/widget/components/viewerUtils.tsx

```diff
@@ -371,49 +372,30 @@ export const getHiliteIds = async (
 };
 
 export const manufactureKeys = async (
-  elementIds: string[],
+  query: string,
   iModelConnection: IModelConnection,
 ): Promise<KeySet> => {
-  // segment ids into batches
-  const batches: string[][] = [];
-  let currBatch: string[] = [];
-  const batchSize = 100000;
-  for (const id of elementIds) {
-    if (currBatch.length < batchSize) {
-      currBatch.push(id);
-    } else {
-      batches.push(currBatch);
-      currBatch = [id];
+  query;
+  iModelConnection;
+  if (query === "") {
+    new KeySet();
+  }
+  const queryWithIdAndECClassName = `SELECT q.ECInstanceId, ec_classname(e.ECClassId) FROM (${query}) q JOIN BisCore.Element e on e.ECInstanceId = q.ECInstanceId`;
+
+  const rowIterator = iModelConnection.query(queryWithIdAndECClassName, undefined, {
+    rowFormat: QueryRowFormat.UseECSqlPropertyIndexes,
+  });
+
+  const keys: InstanceKey[] = [];
+
+  while (true) {
```

Comment:
Rewrite this using the `for await ... of ...` syntax.

Should be something like:
```
for await (const value of rowIterator) {
    keys.push({ id: value[0], className: value[1] });
}
```

Reply:
switched to ```for await``` syntax.